src/textarea/textarea.tsx

```diff
@@ -9,25 +10,120 @@ interface Props {
+                <textarea
+                    id={this.props.id}
+                    className={`form-control text-design ${this.state.stroke} ${
+                        this.props.enableCharacterLimit ? 'character-limited' : ''
+                    } ${this.props.className || ''}`}
+                    rows={this.state.rows}
+                    onBlur={this.props.onBlur}
+                    onChange={this.onChange}
+                    placeholder={this.props.placeholder}
+                    data-qa={`textarea--element-${this.props.id}`}
+                    value={this.props.value}
+                    disabled={this.props.disabled}
+                />
+                {this.props.enableCharacterLimit && (
+                    <CharacterLimitChecker
+                        radius={this.state.radius}
+                        stroke={this.state.stroke}
+                        strokeDasharray={this.state.strokeDasharray}
+                        limit={this.state.limit}
+                    />
+                )}
+            </div>
+        );
+    }
+
+    private onChange(event: React.ChangeEvent<HTMLTextAreaElement>) {
+        this.props.onChange(event.target.value);
+        const currentRows = this.getCurrentRows(event, TextareaConfig.minRows, TextareaConfig.maxRows);
+        this.setState({
+            value: event.target.value,
+            rows: Math.min(currentRows, TextareaConfig.maxRows),
+            limit: this.getLimit(event),
+            stroke: this.isLimitExceeded(event) ? 'error-text' : 'normal-text',
+            strokeDasharray: `${this.getPercentage(event)} 999`,
+        });
+    }
+
+    private getCurrentRows(event: React.ChangeEvent<HTMLTextAreaElement>, minRows: number, maxRows: number) {
+        if (!this.props.enableCharacterLimit) {
+            return this.state.rows;
+        }
+        const textareaLineHeight = 14;
+
+        const previousRows = event.target.rows;
+        event.target.rows = minRows;
+
+        const currentRows = Math.floor(event.currentTarget.scrollHeight / textareaLineHeight);
+
+        if (currentRows === previousRows) {
+            event.target.rows = currentRows;
+        }
+
+        if (currentRows >= maxRows) {
+            event.currentTarget.rows = maxRows;
+            event.currentTarget.scrollTop = event.currentTarget.scrollHeight;
+        }
+        return currentRows;
+    }
+
+    private getPercentage(event: React.ChangeEvent<HTMLTextAreaElement>) {
+        const circleLength = 2 * Math.PI * this.state.radius;
+        let percentageText = (circleLength * this.getLimit(event)) / TextareaConfig.characterLimit;
+        if (this.isLimitExceeded(event)) {
+            percentageText = circleLength;
+        }
+        return percentageText;
+    }
+
+    private isLimitExceeded(event: React.ChangeEvent<HTMLTextAreaElement>) {
+        return this.getLimit(event) <= 0;
+    }
+
+    private getLimit(event: React.ChangeEvent<HTMLTextAreaElement>) {
```

Comment:
This is not the limit but the characters left.

Reply:
fixed